impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -192,19 +256,68 @@ DownsamplePassArgs CalculateDownsamplePassArgs(
   //
   //   !input_snapshot->GetCoverage()->Expand(-local_padding)
   //     .Contains(coverage_hint.value()))
-  Vector2 downsampled_size = source_rect_padded.GetSize() * downsample_scalar;
-  ISize subpass_size =
-      ISize(round(downsampled_size.x), round(downsampled_size.y));
-  Vector2 effective_scalar =
-      Vector2(subpass_size) / source_rect_padded.GetSize();
-
-  Quad uvs = GaussianBlurFilterContents::CalculateUVs(
-      input, snapshot_entity, source_rect_padded, input_snapshot_size);
-  return {
-      .subpass_size = subpass_size,
-      .uvs = uvs,
-      .effective_scalar = effective_scalar,
-  };
+
+  std::optional<Rect> snapshot_coverage = input_snapshot.GetCoverage();
+  if (input_snapshot.transform.IsIdentity() &&
+      source_expanded_coverage_hint.has_value() &&
+      snapshot_coverage.has_value() &&
+      snapshot_coverage->Contains(source_expanded_coverage_hint.value())) {
+    // If the snapshot's transform is the identity transform and we have
+    // coverage hint that fits inside of the snapshots coverage that means the
+    // coverage hint was ignored so we will trim out the area we are interested
+    // in the down-sample pass. This usually means we have a backdrop image
+    // filter.
```

Comment:
Ah, the described behavior happens whenever there's a Texture filter input, right? Makes sense that we need to do some extra work to apply the coverage hint for backdrop filters here.

Reply:
> Ah, the described behavior happens whenever there's a Texture filter input, right?

Yep, exactly.